dali/pipeline/executor/queue_policy.h

```diff
@@ -215,14 +236,18 @@ struct SeparateQueuePolicy {
     return result;
   }
 
-  void ReleaseIdxs(OpType stage, QueueIdxs idxs) {
+  void ReleaseIdxs(OpType stage, QueueIdxs idxs, cudaStream_t stage_stream = 0) {
+    if (idxs[stage] == kInvalidIdx) {
+      return;
+    }
     int current_stage = static_cast<int>(stage);
-    // We have a special case for Support ops - they are set free by a GPU stage,
-    // during QueueOutputIdxs
-    if (stage != OpType::CPU) {
-      if (HasPreviousStage(stage)) {
-        ReleaseStageIdx(PreviousStage(stage), idxs);
-      }
+    // TODO(klecki) when we move to CUDA 10, we should move to cudaLaunchHostFunc
+    if (stage == OpType::MIXED) {
```

Comment:
knitpick: This code is kind of duplicated on line 269. Maybe have a helper or a constructor?

Reply:
Done.